### PR TITLE
[stdlib] Clean up `CPython` primitive type conversion

### PR DESCRIPTION
**Pull request by soraros**
- Align types with the corresponding C API.
- Add function signatures and URLs to the CPython API documentation.
- Reorder methods in the following order: `bool`, `int`, `float`, `str`. Within each type, place `PyT_FromX` before `PyT_AsX`.
- Remove redundant overloads.

CC: @ConnorGray

### CODE REVIEW

**Comment by JoeLoser** [comment]
!sync

**Comment by JoeLoser** [comment]
!sync

File: stdlib/src/python/python_object.mojo [hunk 1]
```diff
@@ -1412,8 +1398,26 @@ struct PythonObject(
...
+        return cpython.PyLong_AsSsize_t(self.py_object)
+
+    fn __float__(self) -> Float64:
+        """Returns a float representation of the object.
+
+        Returns:
+            A floating point value that represents this object.
+        """
+        cpython = _get_global_python_itf().cpython()
+        return cpython.PyFloat_AsDouble(self.py_object)
```
Comment (JoeLoser) [inline stdlib/src/python/python_object.mojo:1411]: **Suggestion** The [docs](https://docs.python.org/3/c-api/float.html#c.PyFloat_AsDouble) show this method may fail. Want to handle this failure handling in a follow-up? The lack-of-failure-handling was pre-existing, so going to merge this as is for now.

Comment (soraros) [reply stdlib/src/python/python_object.mojo:1411]: Indeed we need to add error check to all the python to mojo conversion functions.

**Pull request closed without merging**